### PR TITLE
[MM-39642] - Migrate from gorp to sqlx in store/sqlstore/session_store.go

### PR DESCRIPTION
**Pull request by penthaapatel**
#### Summary

Migrate from `gorp` to `sqlx` in `store/sqlstore/session_store.go`
#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/18880
JIRA - https://mattermost.atlassian.net/browse/MM-39642
#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/session_store.go [hunk 1]
```diff
@@ -58,7 +58,15 @@ func (me SqlSessionStore) Save(session *model.Session) (*model.Session, error) {
 		return nil, err
 	}
 
-	if err := me.GetMaster().Insert(session); err != nil {
+	query, args, err := me.getQueryBuilder().
+		Insert("Sessions").
+		Columns("Id", "Token", "CreateAt", "ExpiresAt", "LastActivityAt", "UserId", "DeviceId", "Roles", "IsOAuth", "ExpiredNotify", "Props").
+		Values(session.Id, session.Token, session.CreateAt, session.ExpiresAt, session.LastActivityAt, session.UserId, session.DeviceId, session.Roles, session.IsOAuth, session.ExpiredNotify, model.MapToJSON(session.Props)).
```
Comment (agnivade) [inline store/sqlstore/session_store.go:64]: For MapToJSON, let's do a json.Marshal manually and return if that throws an error. And then pass `string([]byte)` to the query.

Look at job_store.go/Save

File: store/sqlstore/session_store.go [hunk 2]
```diff
@@ -234,50 +242,51 @@ func (me SqlSessionStore) UpdateLastActivityAt(sessionId string, time int64) err
...
 func (me SqlSessionStore) UpdateProps(session *model.Session) error {
-	_, err := me.GetMaster().Exec(`UPDATE Sessions
-		SET Props=:Props
-		WHERE Id=:Id`, map[string]interface{}{
-		"Props": model.MapToJSON(session.Props),
-		"Id":    session.Id,
-	})
+	query, args, err := me.getQueryBuilder().
+		Update("Sessions").
+		Set("Props", model.MapToJSON(session.Props)).
```
Comment (agnivade) [inline store/sqlstore/session_store.go:268]: Same comment on `MapToJSON`

**Review by agnivade — COMMENTED** [review]
Just some small comments to take care of.

**Review by agnivade — APPROVED** [review]
Thank you @penthaapatel !

**Comment by penthaapatel** [comment]
@agnivade Glad to contribute always! 😄

**Review by isacikgoz — APPROVED** [review]
Apologies for the late review @penthaapatel! LGTM 🎉

**Pull request merged**